Installation issue: chaco on with apple-clang@12:

### Steps to reproduce the issue

```console
$ spack spec -l chaco
Input spec
--------------------------------
chaco

Concretized
--------------------------------
jtdjsxb  chaco@2020-07-16%apple-clang@13.0.0~ipo+shared build_type=RelWithDebInfo arch=darwin-bigsur-skylake
rz3rype      ^cmake@3.22.1%apple-clang@13.0.0~doc+ncurses+ownlibs~qt build_type=Release arch=darwin-bigsur-skylake

```


### Error message

<details><summary>Error message</summary><pre>
>> 80    /var/folders/ck/bwzqcv_s1pz1t9fc86731shh0000gq/T/ptb/spack-stage/spack-s
           tage-chaco-2020-07-16-pq6dhovfy4dswevyeattwcadfrcqltar/spack-src/code/co
           arsen/makecgraph.c:200:2: error: implicit declaration of function 'makec
           coords' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
</pre></details>


### Information on your system

* **Spack:** 0.18.0.dev0 (7367f9e213034ebbf6204c94865aa5edcae784ff)
* **Python:** 3.9.12
* **Platform:** darwin-bigsur-skylake
* **Concretizer:** clingo

### Additional information

_No response_

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/chaco/package.py
+++ b/var/spack/repos/builtin/packages/chaco/package.py
@@ -22,6 +22,8 @@ class Chaco(CMakePackage):
     depends_on('cmake@3.16:', type='build')
 
     def cmake_args(self):
-        return [
-            self.define_from_variant('BUILD_SHARED_LIBS', 'shared')
-        ]
+        opts = [self.define_from_variant('BUILD_SHARED_LIBS', 'shared')]
+        if self.spec.satisfies('%apple-clang@12:'):
+            opts.append(self.define("CMAKE_C_FLAGS",
+                                    "-Wno-error=implicit-function-declaration"))
+        return opts

